design/progress-tracking.md

```diff
@@ -0,0 +1,245 @@
+# Blueprint And Phase Progress Tracking
+
+## Problem Statement
+
+When an actionset triggers a long-running task like the `CopyVolumeData`
+function, the only way to determine any sort of progress is to gain direct access
+to the logs and events of the task pod. This may not always be feasible as the
+user may not have the appropriate RBAC permissions to access these subresource
+endpoints in the pod's namespace. If the export operation takes a long time,
+it's also possible that the pod might be terminated prematurely without leaving
+behind any traces of logs to indicate how far along the task was.
+
+Being able to persist progress data will be very helpful for both live reporting
+of task progress, as well as future retrospection (e.g., the latency of every
+phase of an action).
+
+## Proposed Solution
+
+### Summary
+
+The `ActionSet` CRD's  `status` subresource will be updated with new fields to
+communicate the blueprint and phases' progress status to the user. The
+`kube.ExecOutput()` and `kube.Task()` interfaces will be updated to accept a new
+progress I/O writer.
+
+### Assumptions And Constraints
+
+* The progress computation should not compromise the main data protection task's
+latency nor lead to resource contention. Progress computation will be performed
+on a best-effort basis, where it may be de-prioritized with no guarantee on the
+accuracy of its result, or skipped entirely in the event of resource contention.
+
+### Changes To API
+
+The blueprint's overall progress will be reported under the new
```

Comment:
Similar to Vivek's suggestion, should this say action/actionset's overall progress too?